File: longestCommonPrefix/longComPre.php
class Solution {

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs) {
        if(count($strs)>0){
            $common = $strs[0];
        }else{
            return '';
        }
        for($i=0;$i<strlen($strs[0]);$i++){
            foreach ($strs as $str){
                // if(strlen(strpos($str,$common))==0){
                //     $common = substr($common,0,-1);
                //     break;
                // }
                if(strpos($str,$common)===0){
                    continue;
                }else{
                    $common = substr($common,0,-1);
                    break;
                }
            }
        }
        return $common;
    }
}

//php判断是否等于数字0要用===，或者先判断长度是否为0，为0则是空串
//判断比===快

File: longestCommonPrefix/longComPre.py
class Solution:
    def longestCommonPrefix(self, strs: 'List[str]') -> 'str':
        if len(strs):
            common = strs[0]
        else:
            return ''
        for i in range(len(strs[0])):
            for str in strs[1:]:
                if str.find(common)==0:
                    pass
                else:
                    common=common[0:-1]
                    break
            i+=1
        return common

# 判断数组中后面几串是否从头开始包含第一串，第一串循环去掉最后一位